Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,7 +120,54 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget(target.gameObject, true, out _);
+        }
+
+        /// <summary>
+        /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+        /// (Won't conflict with character attacking enemy or moving on the map)
+        /// </summary>
+        #region Interactables Functions
```

Comment:
pls put all of this in a class InteractLogic

Reply:
Oki !